docs/proposals/push-metrics.md

```diff
@@ -0,0 +1,69 @@
+# Heapster Push Metrics
+
+## Overview and Motivation
+
+Currently, Heapster supports pulling metrics from kubelet, and defines an
+interface for pulling from other sources.  However, in certain cases, it is
+more useful to be able to have services push metrics into Heapster, instead
+of having Heapster pull metrics.
+
+For instance, supporting push metrics makes it easy for cluster admins to add
+custom metrics into Heapster with relatively minimal effort: they can simply
+write a program or script which collects and processes the information, and
+then add a recurring Job or cron job which pushes the metrics (instead of
+having to write a source for Heapster).  This also enables existing tooling
+designed with the push model in mind to be adapted to provide metrics through
+Heapster.
+
+## Design
+
+### API
+
+Metrics added through push metrics will be available as normal through the
+Heapster model.
+
+To add metrics, metrics producers will `POST` new metrics to
+`/api/v1/push/{format}` in the format specified by `{format}`.  Producers will
```

Comment:
How will you exactly attach metrics to pod/containers/namespaces/etc? /api/v1/push seeems to look like a generic bulk push.